### PR TITLE
Discover source maps if not on own last line

### LINKED ISSUE #5058: Sentry should respect sourceMappingURL at EOL
Right now Sentry requires the `sourceMappingURL` directive to be on its own line at the end of a file, which is the literal interpretation of the spec.

But, Chrome and Firefox are more liberal, and will respect `sourceMappingURL` if it follows a bunch of code on the last line of the file, e.g.

```javascript
function add(){for(var a=arguments.length,b=Array(a),c=0;c<a;c++)b[c]=arguments[c];return b.reduce(function(d,e){return d+=e,d},0)}//# sourceMappingURL=app.min.js.map
```

As pointed out by a user, this is easier to produce than first thought:

> However, Webpack's SourceMapDevToolPlugin has an 'append' option, and this requires the dev to remember to add a new line before their sourceMappingURL comment. This is the part that I missed.

Source maps are hard enough as it is; we should probably make this change to be more flexible and make integration easier on users.

cc @maxbittker @mattrobenolt

### PR DESCRIPTION
**Pull request by benvinegar**
Fixes #5058 

cc @mattrobenolt

### CODE REVIEW

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -155,6 +155,19 @@ def discover_sourcemap(result):
                 # We want everything AFTER the indicator, which is 21 chars long
                 sourcemap = line[21:].rstrip()
 
+        # If we still haven't found anything, check end of last line AFTER source code.
+        # This is not the literal interpretation of the spec, but browsers support this.
+        # e.g. {code}#// sourceMappingURL={url}
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: This comment is wrong.

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -155,6 +155,19 @@ def discover_sourcemap(result):
...
+        if not sourcemap:
+            # Only look at last 300 characters to keep search space reasonable (minified
+            # JS on a single line could be tens of thousands of chars). 200 is a totally
+            # arbitrary number / best guess; most sourceMappingURLs are relative and
+            # not very long.
+            search_space = possibilities[-1][-300:].rstrip()
+            match = SOURCE_MAPPING_URL_RE.search(search_space)
+            if match:
+                sourcemap = match.groups()[0]
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: You could alternatively do `match.group(1)`

File: src/sentry/lang/javascript/processor.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ class ZeroReturnError(Exception):
 """, re.X | re.I)
 VERSION_RE = re.compile(r'^[a-f0-9]{32}|[a-f0-9]{40}$', re.I)
 NODE_MODULES_RE = re.compile(r'\bnode_modules/')
-
+SOURCE_MAPPING_URL_RE = re.compile(r'\/\/(?:#|@) ?sourceMappingURL=(.*)$', re.I)
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: Is the `//@` syntax valid here? I know it's definitely legacy and shouldn't be used for anything newer.

Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: I'd say we shouldn't be case insensitive. Everythign else we require an exact `sourceMappingURL` where the case does matter. And back in the day when I looked at source code, browsers were explicitly on casing too.

Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: Should the space after `//#` be optional? We don't allow it to be optional elsewhere.

**Review by mattrobenolt — APPROVED** [review]
All other comments are mostly just suggestions/comments. If you want to change things, sure, but not required.

File: src/sentry/lang/javascript/processor.py (hunk 3 above)
Comment (benvinegar) [inline src/sentry/lang/javascript/processor.py]: > Should the space after //# be optional? We don't allow it to be optional elsewhere.

I suppose I haven't seen anyone complain about this, so I'll remove.

Comment (benvinegar) [inline src/sentry/lang/javascript/processor.py]: > Is the //@ syntax valid here? I know it's definitely legacy and shouldn't be used for anything newer.

Sure.

**Comment by benvinegar** [comment]
@mattrobenolt – addressed all your comments if you want to take a second look.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**